packages/passport/sdk/src/react/PassportContext.tsx

```diff
@@ -0,0 +1,279 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import React, {
+  createContext, useContext, useEffect, useMemo,
+  useState,
+} from 'react';
+import { Web3Provider } from '@ethersproject/providers';
+import { Passport } from '../Passport';
+import { PassportModuleConfiguration, UserProfile } from '../types';
+import { Provider } from '../zkEvm/types';
+
+type PassportContextType = {
+  passportInstance: Passport;
+  isLoggedIn: boolean;
+  isLoading: boolean;
+  error: Error | null;
+  accounts: string[];
+  passportProvider: Provider | null;
+  web3Provider: Web3Provider | null;
+  login: () => Promise<string[]>;
+  logout: () => Promise<void>;
+  loginWithoutWallet: () => Promise<UserProfile | null>;
+  loginWithEthersjs: () => Promise<string[]>;
+  loginCallback: () => Promise<void>;
```

Comment:
Can we have a separate hook for generic Passport functionality (everything in `Passport.ts`), and zkEvm functionality? For a consumer, it's not clear if `accounts` relates to IMX or zkEvm

Reply:
I'm also confused. I have this screenshot attached. They are methods on Passport. Are you saying I should have separate Providers, one for imx and one for zkevm?
![image](https://github.com/user-attachments/assets/5258b289-7ab8-4aca-a920-5f8422cfec0c)

Reply:
Now I think about it. I should just do this for zkevm.

Reply:
I think ideally we'd have three: one for generic Passport (everything in `Passport.ts` except `connectImx` & `connectEvm`), one for exposing IMX functionality, and one for zkEVM. I agree that we can probably disregard the IMX provider